### PR TITLE
Refresh security scans

### PR DESCRIPTION
**Pull request by tatlax3636**
* add dropdown to choose between projects
* add refresh functionality

Future work:
* make more complete/representative tests

### CODE REVIEW

File: proxy.conf.json [hunk 1]
```diff
@@ -10,5 +10,11 @@
       "secure": true,
       "changeOrigin": true,
       "logLevel": "debug"
+    },
+    "/eratocode-security/": {
```
Comment (nameisaravind) [inline proxy.conf.json:14]: @tatlax3636 eratocode is capone specific, use generic endpoint

File: src/app/widget_modules/security-scan/security-scan-detail/security-scan-detail.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  it('should call ngOnInit()', () => {
+    component.ngOnInit();
+  });
+
+  it('should set detailData', () => {
+    const detailData = {
+      clickableHeader: null,
+      clickableContent: null,
+      name: 'eratocode-project',
```
Comment (nameisaravind) [inline src/app/widget_modules/security-scan/security-scan-detail/security-scan-detail.component.spec.ts:37]: @tatlax3636 change this name

**Review by nameisaravind — COMMENTED** [review]
comments added

File: src/app/widget_modules/security-scan/security-scan.service.ts [hunk 3]
```diff
@@ -1,23 +1,33 @@
...
+import {ISecurityScan} from './security-scan-interfaces';
 
 @Injectable({
   providedIn: 'root'
 })
 export class SecurityScanService {
 
-  securityScanDetailEndPoint = '/api/quality/security-analysis';
+  securityScanDetailEndPoint = '/api/quality/security-analysis/ui-widget';
+  baseRefreshUrl = '/eratocode-security/refresh';
```
Comment (nameisaravind) [inline src/app/widget_modules/security-scan/security-scan.service.ts:12]: @tatlax3636 change this endpoint to be generic

File: src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.ts [hunk 4]
```diff
@@ -100,41 +114,141 @@ export class SecurityScanWidgetComponent extends WidgetComponent implements OnIn
...
+    this.securityService.getCodeQuality(this.params.componentId, collectorItemId).subscribe(codeQuality => {
+      this.generateSecurityScanData(codeQuality.result[0], index);
+    });
   }
 
-  generateSecurityScanData(result: ISecurityScan[]) {
-    const sData = result[0].metrics.map(metric => {
+  populateDropdown(collectorItems) {
+    collectorItems.map(item => {
+      item.description = item.description.split(':')[0];
```
Comment (nameisaravind) [inline src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.ts:139]: Plz handle NPE and stringIndex for description @tatlax3636

File: src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.spec.ts [hunk 5]
```diff
@@ -151,14 +216,25 @@ describe('SecurityScanWidgetComponent', () => {
...
     spyOn(dashboardService, 'checkCollectorItemTypeExist').and.returnValues(true, false);
     component.startRefreshInterval();
     component.startRefreshInterval();
     component.startRefreshInterval();
   });
 
-  it('should loadCharts', () => {
-    component.loadCharts([mockSecurityScan]);
-  });
+  // it('should create chart', () => {
```
Comment (nameisaravind) [inline src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.spec.ts:226]: @tatlax3636 delete the commented lines

Comment (tatlax3636) [reply src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.spec.ts:226]: Removed

File: src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.ts (hunk 4 above)
Comment (tatlax3636) [reply src/app/widget_modules/security-scan/security-scan-widget/security-scan-widget.component.ts]: handled for NPE

**Review by nameisaravind — APPROVED** [review]
lgtm

**Pull request merged**